js/services/TorrentSearchEngines/GenericTorrentSearchEngine.js

```diff
@@ -253,7 +254,7 @@ function GenericTorrentSearchEngine(config, $q, $http, $injector) {
                 } else if (config.mirrorResolver && config.mirrorResolver !== null) {
                     $injector.get(config.mirrorResolver).findMirror().then(function(result) {
                         //console.log("Resolved a new working mirror!", result);
-                        mirror = result;
+                        mirror = result; // This doesn't do anything ??
```

Comment:
I have a feeling that this should be `config.mirror = result;`    
TPB is the only SE that has mirror resolution.